pxr/base/arch/attributes.h

```diff
@@ -191,6 +210,15 @@ PXR_NAMESPACE_OPEN_SCOPE
 #   define ARCH_USED_FUNCTION
 #   define ARCH_EMPTY_BASES __declspec(empty_bases)
 
+#if defined(ARCH_SANITIZE_ADDRESS)
+#   define ARCH_NO_SANITIZE_ADDRESS_FUNCTION \
+        __declspec(no_sanitize_address)
+#else
+#   define ARCH_NO_SANITIZE_ADDRESS_FUNCTION
+#endif
+#   define ARCH_NO_SANITIZE_UNDEFINED_FUNCTION
```

Comment:
Given that we don't use `ARCH_NO_SANITIZER_{THREAD,UNDEFINED_BEHAVIOR}_FUNCTION` yet and we don't support them on `msvc`, perhaps we elide them for now (for all compilers)?

Reply:
Good point, really easy to add once support for the corresponding sanitizer is available. Removed but left a comment to serve as information and a place holder